### PR TITLE
[OSDOCS- 8731]:Capture the changes in the steps for OSD cluster deployment on GCP when user selects "Secure boot support for Shielded VMs"

### PR DESCRIPTION
**Pull request by mletalie**
**This PR is for the Shielded VMs for GCP OSD release that is scheduled for 12/14/23. Do not merge until this date.**

Version(s):

4.14+
Issue:

https://issues.redhat.com/browse/OSDOCS-8731
Link to docs preview:

https://68591--docspreview.netlify.app/openshift-dedicated/latest/osd_install_access_delete_cluster/creating-a-gcp-cluster#osd-create-gcp-cluster-ccs_osd-creating-a-cluster-on-gcp
**Go to STEP 8e.**

https://68591--docspreview.netlify.app/openshift-dedicated/latest/osd_install_access_delete_cluster/creating-a-gcp-cluster#osd-create-aws-cluster-ccs_osd-creating-a-cluster-on-gcp
https://68591--docspreview.netlify.app/openshift-dedicated/latest/osd_install_access_delete_cluster/creating-a-gcp-cluster#osd-create-cluster-gcp-account_osd-creating-a-cluster-on-gcp
**Go to STEP 5e.**
QE review:
- [x] QE has approved this change.

Additional information:

image

### CODE REVIEW

**Comment by svmrh** [comment]
Both Step 8.e and 'Important' message looks good. 

This change needs to be added to "Creating a cluster on GCP with Google Cloud Marketplace" and the "Creating a cluster on GCP with a Red Hat cloud account" topics as well. I know you are already working on this, but adding this as a note for others reviewing this PR.

**Comment by mletalie** [comment]
> Can we also add a **_note_** to [Customer Cloud Subscriptions on GCP > Required customer procedure](https://docs.openshift.com/dedicated/osd_planning/gcp-ccs.html#ccs-gcp-customer-procedure_gcp-ccs) topic?
> 
> (feel free to make changes to this sentence)
> 
> * constraints/compute.requireShieldedVm **_(This policy constraint is unsupported only if the cluster is installed without selecting "Enable Secure Boot support for Shielded VMs" on the Cluster details page)_**
> 
> image
> i.e.
> 
> > WARNING
> > To use OpenShift Dedicated in your GCP project, the following GCP organizational policy constraints cannot be in place:
> > 
> > * constraints/iam.allowedPolicyMemberDomains
> > * constraints/compute.restrictLoadBalancerCreationForTypes
> > * constraints/compute.requireShieldedVm **_(This policy constraint is unsupported only if the cluster is installed without selecting "Enable Secure Boot support for Shielded VMs" on the Cluster details page)_**
> > * constraints/compute.vmExternalIpAccess (This policy constraint is unsupported only during installation. You can re-enable the policy constraint after installation.)

Hello @svmrh,
   Thanks for mentioning that issue. The change to that doc is actually being addressed here: 
https://issues.redhat.com/browse/OSDOCS-8065/ https://github.com/openshift/openshift-docs/pull/68506

Based on that ticket's description "_Once the backend allows the creation of clusters that use shieldVM features, the restriction on having the policy "constraints/compute.requireShieldedVM" can be lifted from the documentation_, I removed the line about "constraints/compute.requireShieldedVm" alltogether. However, I can certainly add it back with that descriptive sentence.  

_I do wonder if we add that sentence, should we change the first paragraph to say "To use OpenShift Dedicated in your GCP project, the following GCP organizational policy constraints **are not supported**:_"

Please let me know your thoughts, thanks!

**Comment by mletalie** [comment]
**Reviewers: Please be aware that discussion above regarding "Customer Cloud Subscriptions on GCP > Required customer procedure" topic is not part of the scope of this JIRA/PR and is covered in another ticket as described above. 
Thanks.**

**Comment by mletalie** [comment]
> Both Step 8.e and 'Important' message looks good.
> 
> This change needs to be added to "Creating a cluster on GCP with Google Cloud Marketplace" and the "Creating a cluster on GCP with a Red Hat cloud account" topics as well. I know you are already working on this, but adding this as a note for others reviewing this PR.

Noted: Changes have been added to all three topics.

**Comment by mletalie** [comment]
Hello @yisun, hope you are well! Could I please get a review for this PR:  Please be aware that is the same content applied to the three different topics (it is relevant to all three) so it may seem bigger than it actually is :) 
Thanks!

**Comment by mletalie** [comment]
**FYI: This feature is not to be released until 12/15 (at earliest) so should not be merged until clearance is given by PM.**

File: modules/osd-create-cluster-ccs.adoc [hunk 1]
```diff
@@ -134,6 +134,16 @@ endif::osd-on-gcp[]
 .. Select a cluster version from the *Version* drop-down menu.
 .. Select a cloud provider region from the *Region* drop-down menu.
 .. Select a *Single zone* or *Multi-zone* configuration.
++
+ifdef::osd-on-gcp[]
+.. Optional: Select *Enable Secure Boot for Shielded VMs* to use Shielded VMs when installing your cluster.  For more information, see link:https://cloud.google.com/security/products/shielded-vm[Shielded VMs].
```
Comment (tmalove) [inline modules/osd-create-cluster-ccs.adoc:139]:
```suggestion
.. Optional: Select *Enable Secure Boot for Shielded VMs* to use Shielded VMs when installing your cluster. For more information, see link:https://cloud.google.com/security/products/shielded-vm[Shielded VMs].
```

File: modules/osd-create-cluster-gcp-account.adoc [hunk 2]
```diff
@@ -34,6 +34,16 @@ For more information about service account keys, click the information icon loca
 .. Select a cluster version from the *Version* drop-down menu.
 .. Select a cloud provider region from the *Region* drop-down menu.
 .. Select a *Single zone* or *Multi-zone* configuration.
++
+//ifdef::osd-on-gcp[]
+.. Optional: Select *Enable Secure Boot for Shielded VMs* to use Shielded VMs when installing your cluster.  For more information, see link:https://cloud.google.com/security/products/shielded-vm[Shielded VMs].
```
Comment (tmalove) [inline modules/osd-create-cluster-gcp-account.adoc:39]:
```suggestion
.. Optional: Select *Enable Secure Boot for Shielded VMs* to use Shielded VMs when installing your cluster. For more information, see link:https://cloud.google.com/security/products/shielded-vm[Shielded VMs].
```

File: modules/osd-create-cluster-red-hat-account.adoc [hunk 3]
```diff
@@ -61,6 +61,16 @@ and click *Next*.
 .. Select a cluster version from the *Version* drop-down menu.
 .. Select a cloud provider region from the *Region* drop-down menu.
 .. Select a *Single zone* or *Multi-zone* configuration.
++
+ifdef::osd-on-gcp[]
+.. Optional: Select *Enable Secure Boot for Shielded VMs* to use Shielded VMs when installing your cluster.  For more information, see link:https://cloud.google.com/security/products/shielded-vm[Shielded VMs].
```
Comment (tmalove) [inline modules/osd-create-cluster-red-hat-account.adoc:66]:
```suggestion
.. Optional: Select *Enable Secure Boot for Shielded VMs* to use Shielded VMs when installing your cluster. For more information, see link:https://cloud.google.com/security/products/shielded-vm[Shielded VMs].
```

**Review by tmalove — APPROVED** [review]

File: modules/osd-create-cluster-ccs.adoc [hunk 4]
```diff
@@ -134,6 +134,16 @@ endif::osd-on-gcp[]
 .. Select a cluster version from the *Version* drop-down menu.
 .. Select a cloud provider region from the *Region* drop-down menu.
 .. Select a *Single zone* or *Multi-zone* configuration.
++
+ifdef::osd-on-gcp[]
+.. Optional: Select *Enable Secure Boot for Shielded VMs* to use Shielded VMs when installing your cluster. For more information, see link:https://cloud.google.com/security/products/shielded-vm[Shielded VMs].
++
+[IMPORTANT]
+====
+To successfully create a cluster, you must select *Enable Secure Boot support for Shielded VMs* if your organization has the policy constraint 'constraints/compute.requireShieldedVm' enabled. For more information regarding GCP organizational policy constraints, see link:https://cloud.google.com/resource-manager/docs/organization-policy/org-policy-constraints[Organization policy constraints].
```
Comment (adellape) [inline modules/osd-create-cluster-ccs.adoc:143]: "constraints/compute.requireShieldedVm" should be marked up somehow, either backticks or maybe bold. Looking elsewhere in the repo, it seems that other policy constraints have used backticks:

```suggestion
To successfully create a cluster, you must select *Enable Secure Boot support for Shielded VMs* if your organization has the policy constraint `constraints/compute.requireShieldedVm` enabled. For more information regarding GCP organizational policy constraints, see link:https://cloud.google.com/resource-manager/docs/organization-policy/org-policy-constraints[Organization policy constraints].
```

There are 2 other instances of this in the PR that should likewise be marked up.

File: modules/osd-create-cluster-gcp-account.adoc [hunk 5]
```diff
@@ -34,6 +34,16 @@ For more information about service account keys, click the information icon loca
 .. Select a cluster version from the *Version* drop-down menu.
 .. Select a cloud provider region from the *Region* drop-down menu.
 .. Select a *Single zone* or *Multi-zone* configuration.
++
+//ifdef::osd-on-gcp[]
+.. Optional: Select *Enable Secure Boot for Shielded VMs* to use Shielded VMs when installing your cluster. For more information, see link:https://cloud.google.com/security/products/shielded-vm[Shielded VMs].
++
+[IMPORTANT]
+====
+To successfully create a cluster, you must select *Enable Secure Boot support for Shielded VMs* if your organization has the policy constraint 'constraints/compute.requireShieldedVm' enabled. For more information regarding GCP organizational policy constraints, see link:https://cloud.google.com/resource-manager/docs/organization-policy/org-policy-constraints[Organization policy constraints].
```
Comment (adellape) [inline modules/osd-create-cluster-gcp-account.adoc:43]: Same here re: markup the constraint.

File: modules/osd-create-cluster-red-hat-account.adoc [hunk 6]
```diff
@@ -61,6 +61,16 @@ and click *Next*.
 .. Select a cluster version from the *Version* drop-down menu.
 .. Select a cloud provider region from the *Region* drop-down menu.
 .. Select a *Single zone* or *Multi-zone* configuration.
++
+ifdef::osd-on-gcp[]
+.. Optional: Select *Enable Secure Boot for Shielded VMs* to use Shielded VMs when installing your cluster. For more information, see link:https://cloud.google.com/security/products/shielded-vm[Shielded VMs].
++
+[IMPORTANT]
+====
+To successfully create a cluster, you must select *Enable Secure Boot support for Shielded VMs* if your organization has the policy constraint 'constraints/compute.requireShieldedVm' enabled. For more information regarding GCP organizational policy constraints, see link:https://cloud.google.com/resource-manager/docs/organization-policy/org-policy-constraints[Organization policy constraints].
```
Comment (adellape) [inline modules/osd-create-cluster-red-hat-account.adoc:70]: Same here re: markup the constraint.

File: modules/osd-create-cluster-gcp-account.adoc [hunk 7]
```diff
@@ -34,6 +34,16 @@ For more information about service account keys, click the information icon loca
 .. Select a cluster version from the *Version* drop-down menu.
 .. Select a cloud provider region from the *Region* drop-down menu.
 .. Select a *Single zone* or *Multi-zone* configuration.
++
+//ifdef::osd-on-gcp[]
```
Comment (adellape) [inline modules/osd-create-cluster-gcp-account.adoc:38]: Is the commenting-out here just placeholder for future uncommenting, or could the `ifdef`/`endif` be omitted for now?

**Review by adellape — APPROVED** [review]

**Pull request merged**